server/src/main/java/com/paypal/selion/grid/servlets/transfer/DefaultCriteria.java

```diff
@@ -0,0 +1,118 @@
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets.transfer;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.paypal.selion.grid.servlets.transfer.DefaultManagedArtifact.DefaultRequestParameters;
+import com.paypal.selion.logging.SeLionGridLogger;
+
+/**
+ * {@link Criteria} to match a {@link DefaultManagedArtifact} uniquely. Criteria uses artifact name, user id and
+ * artifact folder to uniquely identify a {@link DefaultManagedArtifact}. Parameters artifactName, artifactFolder and
+ * userId match artifact name, folder name and parent folder name of some {@link DefaultManagedArtifact}
+ * respectively.
+ */
+final class DefaultCriteria implements Criteria {
+    
+    private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(DefaultCriteria.class);
+
+    private String artifactName;
+
+    private String uuidFolderName;
+
+    private String subFolderName;
+
+    public DefaultCriteria(Map<String, String> parametersMap) {
+        validateParametersMap(parametersMap);
+        this.artifactName = parametersMap.get(ManagedArtifact.ARTIFACT_FILE_NAME);
+        this.uuidFolderName = parametersMap.get(DefaultRequestParameters.USER_ID);
+        this.subFolderName = parametersMap.get(ManagedArtifact.ARTIFACT_FOLDER_NAME);
+    }
+
+    private void validateParametersMap(Map<String, String> parametersMap) {
+        if (!parametersMap.containsKey(ManagedArtifact.ARTIFACT_FILE_NAME)
+                || !parametersMap.containsKey(DefaultRequestParameters.USER_ID)) {
+            throw new ArtifactDownloadException("Request missing essential parameters: "
+                    + ManagedArtifact.ARTIFACT_FILE_NAME + ", " + DefaultRequestParameters.USER_ID);
+        }
+    }
+
+    public String getArtifactName() {
+        return artifactName;
+    }
+
+    public String getUUIDFolderName() {
+        return uuidFolderName;
+    }
+
+    public String getArtifactFolder() {
+        return subFolderName;
+    }
+
+    public Map<String, String> asMap() {
+        LOGGER.entering();
+        Map<String, String> contentMap = new HashMap<>();
+        contentMap.put(ManagedArtifact.ARTIFACT_FILE_NAME, getArtifactName());
+        contentMap.put(DefaultRequestParameters.USER_ID, getUUIDFolderName());
+        if (!StringUtils.isBlank(getArtifactFolder())) {
```

Comment:
This was pre-existing and I believe the if check is necessary, since `artifactFolder` is an optional parameter in the `DefaultManagedArtifact` implementation, though I need to see how `asMap()` is used.